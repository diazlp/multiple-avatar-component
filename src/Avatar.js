import PropTypes from 'prop-types';
import getSizeClass from './utils/getSizeClass';

const Avatar = ({ name, size, dataTestId }) => {
  const initials = isNaN(+name) && name.split(' ').map(part => part[0]).join('').toUpperCase();

  return (
    <div data-testid={dataTestId || `avatar`} className={`avatar bg-gray-500 ${getSizeClass(size)} flex items-center justify-center`}>
      <span className="text-white font-bold">{initials || `+${name}`}</span>
    </div>
  );
};

Avatar.propTypes = {
  name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
  dataTestId: PropTypes.any
};

Avatar.defaultProps = {
  name: 'Diaz Linggaputra',
  size: 'md',
};

export default Avatar;