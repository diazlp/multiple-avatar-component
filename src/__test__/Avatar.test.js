import { render, screen, cleanup } from '@testing-library/react';
import Avatar from '../Avatar';

beforeEach(() => {
  cleanup();
});

describe('Avatar Component Testing', () => {
  it('renders default component when passed no props', () => {
    render(<Avatar />);
    const initials = screen.getByText('DL');
    const avatar = screen.getByTestId('avatar');
    expect(initials).toBeInTheDocument();
    expect(avatar).toHaveClass('w-12 h-12 text-base');
  });

  it('renders the initials of the name', () => {
    render(<Avatar name="John Doe" />);
    const initials = screen.getByText('JD');
    expect(initials).toBeInTheDocument();
  });

  it('renders the avatar name as number for hidden names count', () => {
    render(<Avatar name={10} />);
    const name = screen.getByText('+10');
    expect(name).toBeInTheDocument();
  });

  it('applies the correct class if given xs as size props', () => {
    render(<Avatar name="John Doe" size="xs" />);
    const avatar = screen.getByTestId('avatar');
    expect(avatar).toHaveClass('w-8 h-8 text-xs');
  });

  it('applies the correct class if given sm as size props', () => {
    render(<Avatar name="John Doe" size="sm" />);
    const avatar = screen.getByTestId('avatar');
    expect(avatar).toHaveClass('w-10 h-10 text-sm');
  });

  it('applies the correct class if given md as size props', () => {
    render(<Avatar name="John Doe" size="md" />);
    const avatar = screen.getByTestId('avatar');
    expect(avatar).toHaveClass('w-12 h-12 text-base');
  });

  it('applies the correct class if given lg as size props', () => {
    render(<Avatar name="John Doe" size="lg" />);
    const avatar = screen.getByTestId('avatar');
    expect(avatar).toHaveClass('w-16 h-16 text-lg');
  });

  it('matches component snapshot', () => {
    const { container } = render(<Avatar name="John Doe" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});