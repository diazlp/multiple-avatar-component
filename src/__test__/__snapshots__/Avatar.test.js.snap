// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component Testing matches component snapshot 1`] = `
<div
  class="avatar bg-gray-500 w-12 h-12 text-base flex items-center justify-center"
  data-testid="avatar"
>
  <span
    class="text-white font-bold"
  >
    JD
  </span>
</div>
`;
